tests/components/caldav/test_calendar.py

```diff
@@ -1140,3 +1200,63 @@ async def test_config_entry_supported_components(
     # No entity created when no components exist
     state = hass.states.get("calendar.calendar_4")
     assert not state
+
+
+@pytest.mark.parametrize("tz", [UTC])
+async def test_add_vevent(
+    hass: HomeAssistant,
+    setup_platform_cb: Callable[[], Awaitable[None]],
+    calendars: list[Mock],
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Test adding a VEVENT to the calendar."""
+    await setup_platform_cb()
+    called = {}
+
+    def fake_add_event(ics_data: str) -> None:
+        called["ics_data"] = ics_data
+
+    monkeypatch.setattr(calendars[0], "add_event", fake_add_event)
```

Comment:
Is this just this?
```
calendars[0].add_event = fake_add_event
```

But really you're just trying to get the call arguement to this mock? So just get the call args: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args

Similarly below, just check the number of times the mock was called?

Reply:
If i understand correctly form docs: 
`monkeypatch.setattr` makes temporary change which exist only for time of the test and reverts objects to original state after test. 
If i do `calendars[0].add_event = fake_add_event` I will override the object globally which _could_ affect other tests as `Mock` are mutable.


In this test i'm trying to check if ICS (ICalendar) was generated correctly (assuming implementation is black box, even tho in reality it checks functions and assigning variables of ical library).  
If you think different approach would be better please let me know :)